riscv/mmu.h

```diff
@@ -87,11 +87,18 @@ class mmu_t
   #define load_func(type, prefix, xlate_flags) \
     inline type##_t prefix##_##type(reg_t addr, bool require_alignment = false) { \
       const size_t size = sizeof(type##_t); \
+      const reg_t vpn = addr >> PGSHIFT; \
+      if ((xlate_flags) == 0 && unlikely(tlb_load_tag[vpn % TLB_ENTRIES] == (vpn | TLB_CHECK_TRIGGERS))) { \
+        if (!matched_trigger) { \
+          matched_trigger = check_trigger_address_before(triggers::OPERATION_LOAD, addr); \
+          if (matched_trigger) \
+            throw *matched_trigger; \
+        } \
```

Comment:
Both times this new method is called, the return value is immediately thrown. Why not `throw` from within the `check_trigger_address_before()` function?

Reply:
Good point. Done.

Reply:
Now that you've made me aware of [this rather crazy system](https://github.com/riscv-software-src/riscv-isa-sim/blob/6043dab46b4f1c4f7fd2d9fe4c03604e6b414066/riscv/execute.cc#L314-L326), I'm no longer sure this is a good idea. Don't we need to set `mmu->matched_trigger` sometimes before throwing?

Reply:
Good point. I missed this because it's only relevant when timing=after, which isn't covered by the spike tests and OpenOCD doesn't use it either.

On my todo list is to make both timing=before and timing=after work correctly in spike. When I get around to that I'll also revisit this logic, which seems suspect but currently harmless.